homeassistant/components/workday/binary_sensor.py

```diff
@@ -165,17 +174,45 @@ async def async_setup_entry(
         _holiday_string = holiday_date.strftime("%Y-%m-%d")
         LOGGER.debug("%s %s", _holiday_string, name)
 
+    entity = IsWorkdaySensor(
+        obj_holidays,
+        workdays,
+        excludes,
+        days_offset,
+        sensor_name,
+        entry.entry_id,
+    )
+
+    async def check_date(service_call: ServiceCall) -> ServiceResponse:
+        """Check if date is workday or not."""
+        date_to_test: date = service_call.data[CHECK_DATE]
+        holiday_date = (
+            date_to_test in entity._obj_holidays  # pylint: disable=protected-access
+        )
+
+        date_string = date_to_test.strftime("%Y-%m-%d")
+
+        return {"date": date_string, "workday": not holiday_date}
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_CHECK_DATE,
+        check_date,
+        cv.make_entity_service_schema({vol.Required(CHECK_DATE): cv.date}),
+        SupportsResponse.ONLY,
+    )
+
+    entity = IsWorkdaySensor(
+        obj_holidays,
+        workdays,
+        excludes,
+        days_offset,
+        sensor_name,
+        entry.entry_id,
+    )
```

Comment:
I noticed that this is a duplicate of the assignment on line 177. Is that intentional?